core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/rules/SnowflakeCallMapper.scala

```diff
@@ -118,4 +143,124 @@ class SnowflakeCallMapper extends ir.CallMapper with ir.IRHelpers {
       throw TranspileException(
         s"Wrong number of arguments to SPLIT_PART, expected 3, got ${other.size}: ${other.mkString(", ")}")
   }
+
+  private def regexpExtract(args: Seq[ir.Expression]): ir.Expression = {
+    if (args.size == 2) {
+      ir.RegExpExtract(args.head, args(1), ir.Literal(1))
+    } else if (args.size == 3) {
+      ir.RegExpExtract(args.head, args(1), args(2))
+    } else {
+      throw TranspileException(s"WRONG number of arguments to REGEXP_EXTRACT, expected 2 or 3, got ${args.size}")
+    }
+  }
+
+  private def translateDateOrTimePart(input: ir.Expression): String = input match {
```

Comment:
don't you want to put date fn conversions into a separate object perhaps?

Reply:
That's a good idea, `translateDateOrTimePart` could be the only public method of `SnowflakeTimeUnit` (which would then need a better name). I'll do that.